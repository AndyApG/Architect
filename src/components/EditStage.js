import React from 'react';

const StageTitle = () => null;
const StageNodeType = () => null;
const StageForm = () => null;
const StagePrompts = () => null;
const StagePanels = () => null;

const EditStage = () => (
  <div className="edit-stage">
    <div className="edit-stage__section">
      <StageTitle />
      <StageNodeType />
      <StageForm />
      <StagePrompts />
      <StagePanels />
    </div>
    <div className="edit-skip-logic__guidance">
      This is some guidance about that section.
    </div>
  </div>
);

export default EditStage;
