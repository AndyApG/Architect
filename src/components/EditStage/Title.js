/* eslint-disable */
import React from 'react';
import { Button, SeamlessTextInput } from '../../components/Form';

const Title = ({ title, onChange }) => (
  <div className="edit-stage__title">
    <SeamlessTextInput value={title} onChange={onChange} />
    <Button size="small">Tiggle code view</Button>
  </div>
);

export default Title;
