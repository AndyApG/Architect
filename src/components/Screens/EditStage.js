import React, { PureComponent } from 'react';
import PropTypes from 'prop-types';
import { connect } from 'react-redux';
import { bindActionCreators } from 'redux';
import { compose } from 'recompose';
import {
  submit as submitForm,
  isDirty as isFormDirty,
  isInvalid as isFormInvalid,
} from 'redux-form';
import { has, find } from 'lodash';
import { Button } from '../../ui/components';
import Card from './ProtocolCard';
import StageEditor from '../../components/StageEditor';
import { getProtocol } from '../../selectors/protocol';
import { actionCreators as stageActions } from '../../ducks/modules/protocol/stages';
import { actionCreators as previewActions } from '../../ducks/modules/preview';

const formName = 'edit-stage';

class EditStage extends PureComponent {
  static propTypes = {
    dirty: PropTypes.bool.isRequired,
    invalid: PropTypes.bool.isRequired,
    show: PropTypes.bool.isRequired,
    submitForm: PropTypes.func.isRequired,
    onComplete: PropTypes.func,
    stage: PropTypes.object.isRequired,
    id: PropTypes.oneOfType([PropTypes.string, PropTypes.number]),
    insertAtIndex: PropTypes.number,
    updateStage: PropTypes.func.isRequired,
    createStage: PropTypes.func.isRequired,
    previewStage: PropTypes.func.isRequired,
    closePreview: PropTypes.func.isRequired,
    transitionState: PropTypes.string,
  };

  static defaultProps = {
    id: null,
    insertAtIndex: null,
    onComplete: () => {},
    transitionState: null,
  };

  get isDirty() {
    return this.props.dirty || !has(this.props.stage, 'id');
  }

  get buttons() {
    return this.isDirty ? [
      <Button
        key="continue"
        onClick={this.props.submitForm}
      >Continue</Button>,
    ] : [];
  }

  get secondaryButtons() {
    return [
      <Button
        key="preview"
        onClick={this.handlePreview}
        color="paradise-pink"
        disabled={this.props.invalid}
      >Preview</Button>,
    ];
  }

  handleComplete = () => {
    this.props.closePreview(); // Hide preview
    this.props.onComplete();
  }

  handlePreview = () => this.props.previewStage();

  handleSubmit = (stage) => {
    const { id, insertAtIndex } = this.props;

    if (id) {
      this.props.updateStage(id, stage);
    } else {
      this.props.createStage(stage, insertAtIndex);
    }

    this.handleComplete();
  }

  handleCancel = this.handleComplete;

  render() {
    const { stage, show, transitionState } = this.props;

    return (
      <Card
        buttons={this.buttons}
        secondaryButtons={this.secondaryButtons}
        show={show}
        transitionState={transitionState}
        onCancel={this.handleCancel}
      >
        <StageEditor
          stage={stage}
          form={formName}
          onSubmit={this.handleSubmit}
          previewStage={this.props.previewStage}
        />
      </Card>
    );
  }
}

const mapStateToProps = (state, props) => {
  const protocol = getProtocol(state);
  const stage = find(protocol.stages, ['id', props.id]) || { type: props.type };

  return ({
    stage,
    dirty: isFormDirty(formName)(state),
    invalid: isFormInvalid(formName)(state),
  });
};

const mapDispatchToProps = (dispatch, props) => {
  const stageMeta = {
    id: props.id,
  };

  return {
    submitForm: () => dispatch(submitForm(formName)),
    updateStage: bindActionCreators(stageActions.updateStage, dispatch),
    createStage: bindActionCreators(stageActions.createStage, dispatch),
    closePreview: bindActionCreators(previewActions.closePreview, dispatch),
    previewStage: () => dispatch(previewActions.previewStageByFormName(stageMeta, formName)),
  };
};

export { EditStage };

export default compose(
  connect(mapStateToProps, mapDispatchToProps),
)(EditStage);
