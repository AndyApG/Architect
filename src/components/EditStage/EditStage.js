/* eslint-disable */
import React from 'react';
import Title from './Title';

const StageNodeType = () => null;
const StageForm = () => null;
const StagePrompts = () => null;
const StagePanels = () => null;

const EditStage = ({ stage, onChange }) => (
  <div className="edit-stage">
    <div className="edit-stage__section">
      <Title title={stage.title} onChange={title => { onChange({ title }); } }/>
      <StageNodeType />
      <StageForm />
      <StagePrompts />
      <StagePanels />
    </div>
    <div className="edit-skip-logic__guidance">
      This is some guidance about that section.
    </div>
  </div>
);

export default EditStage;
