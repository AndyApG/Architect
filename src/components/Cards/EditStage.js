import React, { PureComponent } from 'react';
import PropTypes from 'prop-types';
import { connect } from 'react-redux';
import { bindActionCreators } from 'redux';
import {
  submit as submitForm,
  isDirty as isFormDirty,
  isInvalid as isFormInvalid,
} from 'redux-form';
import { pick, has } from 'lodash';
import { makeGetStage } from '../../selectors/protocol';
import { Button } from '../../ui/components';
import { ProtocolCard } from '../../containers/ProtocolCard';
import StageEditor from '../../components/StageEditor';
import { actionCreators as stageActions } from '../../ducks/modules/protocol/stages';

class EditStage extends PureComponent {
  static propTypes = {
    dirty: PropTypes.bool.isRequired,
    invalid: PropTypes.bool.isRequired,
    continue: PropTypes.func.isRequired,
    onComplete: PropTypes.func.isRequired,
    stage: PropTypes.object.isRequired,
    stageId: PropTypes.oneOfType([PropTypes.string, PropTypes.number]),
    type: PropTypes.string,
    insertAtIndex: PropTypes.number,
    updateStage: PropTypes.func.isRequired,
    createStage: PropTypes.func.isRequired,
  };

  static defaultProps = {
    stageId: null,
    insertAtIndex: null,
    type: null,
  };

  onUpdate = (stage) => {
    const { stageId, insertAtIndex } = this.props;

    if (stageId) {
      this.props.updateStage(stageId, stage);
    } else {
      this.props.createStage(stage, insertAtIndex);
    }

    this.props.onComplete();
  }

  get isDirty() {
    return this.props.dirty || !has(this.props.stage, 'id');
  }

  renderButtons() {
    return [].concat(
      this.isDirty ? [<Button key="continue" size="small" disabled={this.props.invalid} onClick={this.props.continue}>Continue</Button>] : [],
    );
  }

  render() {
    const { stage } = this.props;

    return (
      <ProtocolCard
        buttons={this.renderButtons()}
        {...pick(this.props, ['show', 'className', 'onCancel'])}
      >
        <StageEditor
          stage={stage}
          onSubmit={this.onUpdate}
        />
      </ProtocolCard>
    );
  }
}

const makeMapStateToProps = () => {
  const getStage = makeGetStage();

  return (state, props) => {
    const stage = getStage(state, props) || { type: props.type };

    return ({
      stage,
      dirty: isFormDirty('edit-stage')(state),
      invalid: isFormInvalid('edit-stage')(state),
    });
  };
};

const mapDispatchToProps = dispatch => ({
  continue: () => dispatch(submitForm('edit-stage')),
  updateStage: bindActionCreators(stageActions.updateStage, dispatch),
  createStage: bindActionCreators(stageActions.createStage, dispatch),
});

export { EditStage };

export default connect(makeMapStateToProps, mapDispatchToProps)(EditStage);
