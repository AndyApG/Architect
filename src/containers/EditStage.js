import React, { PureComponent } from 'react';
import { connect } from 'react-redux';
import { bindActionCreators, compose } from 'redux';
import PropTypes from 'prop-types';
import { Button } from 'network-canvas-ui';
import { makeGetStage } from '../selectors/protocol';
import { actionCreators as stageActions } from '../ducks/modules/stages';
import { ProtocolCard } from '../containers/ProtocolCard';
import EditStage from '../components/EditStage';
import { Draft } from '../behaviours';

const defaultStage = {
};

class EditStageContainer extends PureComponent {
  static propTypes = {
    hasChanges: PropTypes.bool,
    stageId: PropTypes.number,
    onComplete: PropTypes.func,
    draft: PropTypes.any.isRequired,
    updateStage: PropTypes.func.isRequired,
    updateDraft: PropTypes.func.isRequired,
  };

  static defaultProps = {
    draft: null,
    stageId: null,
    hasChanges: false,
    onComplete: () => {},
  }

  onSave = () => {
    const stageId = this.props.stageId;

    this.props.updateStage(
      stageId,
      {
        ...this.props.draft,
      },
    );

    this.props.onComplete();
  };

  renderButtons() {
    return [].concat(
      this.props.hasChanges ? [<Button key="save" size="small" onClick={this.onSave}>Save</Button>] : [],
    );
  }

  render() {
    const {
      hasChanges,
      stageId,
      onComplete,
      draft,
      updateStage,
      updateDraft,
      ...rest
    } = this.props;

    return (
      <ProtocolCard
        title="Edit Stage"
        buttons={this.renderButtons()}
        {...rest}
      >
        <EditStage
          stage={draft}
          onChange={this.props.updateDraft}
        />
      </ProtocolCard>
    );
  }
}

function makeMapStateToProps() {
  const getStage = makeGetStage();

  return function mapStateToProps(state, props) {
    const stage = getStage(state, props) || defaultStage;
    return { draft: stage };
  };
}

function mapDispatchToProps(dispatch) {
  return {
    updateStage: bindActionCreators(stageActions.updateStage, dispatch),
  };
}

export { EditStageContainer as EditStage };

export default compose(
  connect(makeMapStateToProps, mapDispatchToProps),
  Draft,
)(EditStageContainer);
