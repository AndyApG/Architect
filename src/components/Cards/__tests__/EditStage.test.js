/* eslint-env jest */

import React from 'react';
import { shallow } from 'enzyme';
import { EditStage } from '../EditStage';

const mockProps = {
  dirty: false,
  invalid: false,
  show: true,
  submitForm: () => {},
  onComplete: () => {},
  stage: {},
  updateStage: () => {},
  createStage: () => {},
  previewStage: () => {},
  closePreview: () => {},
};

describe('<EditStage />', () => {
  it('can render', () => {
    const component = shallow(<EditStage {...mockProps} />);

    expect(component).toMatchSnapshot();
  });
});
