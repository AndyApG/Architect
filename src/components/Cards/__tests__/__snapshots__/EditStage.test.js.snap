// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<EditStage /> can render 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <EditStage
    createStage={[Function]}
    dirty={false}
    insertAtIndex={null}
    invalid={false}
    onComplete={[Function]}
    previewStage={[Function]}
    show={true}
    stage={Object {}}
    stageId={null}
    submitForm={[Function]}
    updateStage={[Function]}
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "class",
    "props": Object {
      "buttons": Array [
        <Button
          color=""
          content=""
          icon=""
          iconPosition="right"
          onClick={[Function]}
          size=""
        >
          Continue
        </Button>,
      ],
      "children": <defaultProps(Connect(withState(withHandlers(ReduxForm))))
        form="edit-stage"
        onSubmit={[Function]}
        previewStage={[Function]}
        stage={Object {}}
      />,
      "onCancel": [Function],
      "secondaryButtons": Array [
        <Button
          color="paradise-pink"
          content=""
          icon=""
          iconPosition="right"
          onClick={[Function]}
          size=""
        >
          Preview
        </Button>,
      ],
      "show": true,
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "function",
      "props": Object {
        "form": "edit-stage",
        "onSubmit": [Function],
        "previewStage": [Function],
        "stage": Object {},
      },
      "ref": null,
      "rendered": null,
      "type": [Function],
    },
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "buttons": Array [
          <Button
            color=""
            content=""
            icon=""
            iconPosition="right"
            onClick={[Function]}
            size=""
          >
            Continue
          </Button>,
        ],
        "children": <defaultProps(Connect(withState(withHandlers(ReduxForm))))
          form="edit-stage"
          onSubmit={[Function]}
          previewStage={[Function]}
          stage={Object {}}
        />,
        "onCancel": [Function],
        "secondaryButtons": Array [
          <Button
            color="paradise-pink"
            content=""
            icon=""
            iconPosition="right"
            onClick={[Function]}
            size=""
          >
            Preview
          </Button>,
        ],
        "show": true,
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "form": "edit-stage",
          "onSubmit": [Function],
          "previewStage": [Function],
          "stage": Object {},
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
